openmetadata-ui/src/main/resources/ui/src/utils/EntityLineageUtils.tsx

```diff
@@ -515,3 +528,181 @@ export const getNodeRemoveButton = (onClick: () => void) => {
+    if (curr.toColumn === data.data?.targetHandle) {
+      const newCol = {
+        ...curr,
+        fromColumns:
+          curr.fromColumns?.filter((c) => c !== data.data?.sourceHandle) || [],
+      };
+      if (newCol.fromColumns?.length) {
+        return [...col, newCol];
+      } else {
+        return col;
+      }
+    }
+
+    return [...col, curr];
+  }, [] as ColumnLineage[]);
+
+  return {
+    sqlQuery: lineageDetails.sqlQuery || '',
+    columnsLineage: columnsLineage,
+  };
+};
+
+export const getUpdatedUpstreamDownStreamEdgeArr = (
+  edge: EntityLineageEdge[],
+  data: SelectedEdge,
+  lineageDetails: LineageDetails
+) => {
+  return edge.map((down) => {
+    if (
+      down.fromEntity === data.source.id &&
+      down.toEntity === data.target.id
+    ) {
+      return {
+        ...down,
+        lineageDetails: lineageDetails,
+      };
+    }
+
+    return down;
+  });
+};
+
+export const getRemovedNodeData = (
+  nodes: EntityReference[],
+  edge: Edge,
+  entity: EntityReference,
+  selectedEntity: EntityReference
+) => {
+  let targetNode = nodes.find((n) => edge.target?.includes(n.id));
+  let sourceNode = nodes.find((n) => edge.source?.includes(n.id));
+  const selectedNode = isEmpty(selectedEntity) ? entity : selectedEntity;
+
+  if (isUndefined(targetNode)) {
+    targetNode = selectedNode;
+  }
+  if (isUndefined(sourceNode)) {
+    sourceNode = selectedNode;
+  }
+
+  return {
+    id: edge.id,
+    source: sourceNode,
+    target: targetNode,
+  };
+};
+
+const getSourceNode = (
+  edge: EntityLineageEdge[],
+  queryStr: string | null,
+  id: string
+) => {
+  return edge.find(
+    (d) =>
+      (queryStr?.includes(d.fromEntity) || queryStr?.includes(d.toEntity)) &&
+      queryStr !== id
+  );
+};
+
+export const getStreamInfo = (
+  updatedLineageData: EntityLineage,
+  params: Edge | Connection
+) => {
+  const { entity } = updatedLineageData;
+  const { target, source } = params;
+  const sourceDownstreamNode = getSourceNode(
+    updatedLineageData.downstreamEdges || [],
+    source,
+    entity.id
+  );
+
+  const sourceUpStreamNode = getSourceNode(
+    updatedLineageData.upstreamEdges || [],
+    source,
+    entity.id
+  );
+
+  const targetDownStreamNode = getSourceNode(
+    updatedLineageData.downstreamEdges || [],
+    target,
+    entity.id
+  );
+
+  const targetUpStreamNode = getSourceNode(
+    updatedLineageData.upstreamEdges || [],
+    target,
+    entity.id
+  );
+
+  const isUpstream =
+    (!isNil(sourceUpStreamNode) && !isNil(targetDownStreamNode)) ||
+    !isNil(sourceUpStreamNode) ||
+    !isNil(targetUpStreamNode) ||
+    target?.includes(entity.id);
+
+  const isDownstream =
+    (!isNil(sourceDownstreamNode) && !isNil(targetUpStreamNode)) ||
+    !isNil(sourceDownstreamNode) ||
+    !isNil(targetDownStreamNode) ||
+    source?.includes(entity.id);
+
+  if (isUpstream) {
+    return 'upstream';
+  } else if (isDownstream) {
+    return 'downstream';
+  }
```

Comment:
We have two types of nodes `upstream` and `downstream` so can we create an enum for that and use it in all the places?